x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/EsqlArithmeticOperation.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic;
+
+import org.elasticsearch.common.TriFunction;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.compute.operator.EvalOperator;
+import org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator;
+import org.elasticsearch.xpack.esql.EsqlIllegalArgumentException;
+import org.elasticsearch.xpack.esql.evaluator.mapper.EvaluatorMapper;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Cast;
+import org.elasticsearch.xpack.esql.type.EsqlDataTypeRegistry;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.ArithmeticOperation;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.BinaryArithmeticOperation;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataType;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.xpack.ql.type.DataTypes.DOUBLE;
+import static org.elasticsearch.xpack.ql.type.DataTypes.INTEGER;
+import static org.elasticsearch.xpack.ql.type.DataTypes.LONG;
+import static org.elasticsearch.xpack.ql.type.DataTypes.UNSIGNED_LONG;
+
+abstract class EsqlArithmeticOperation extends ArithmeticOperation implements EvaluatorMapper {
+
+    /**
+     * The only role of this enum is to fit the super constructor that expects a BinaryOperation which is
+     * used just for its symbol.
+     * The rest of the methods should not be triggered hence the UOE.
+     */
+    enum OperationSymbol implements BinaryArithmeticOperation {
+        ADD("+"),
+        SUB("-"),
+        MUL("*"),
+        DIV("/"),
+        MOD("%");
+
+        private final String symbol;
+
+        OperationSymbol(String symbol) {
+            this.symbol = symbol;
+        }
+
+        @Override
+        public String getWriteableName() {
+            throw new UnsupportedOperationException();
```

Comment:
question: even if this should be unreachable, didn't we want to go with exceptions that can be handled gracefully by the server (in case we end up here, after all)? Of course, for now there's only `EsqlIllegalArgumentException`, which does not really describe this correctly; I'll probably add `EqslUnsupportedOperationException` [in my PR](https://github.com/elastic/elasticsearch/pull/98599) to remedy this.

Reply:
After some tinkering I also found out that the server seems to handle UOE just like EIAE and returns a proper 500 response; it only really doesn't like proper errors, which crash the server (as expected). So, I agree that UOE is perfectly fine here, and EIAE would make no sense.